src/main/ruby/truffleruby/core/process.rb

```diff
@@ -729,40 +731,41 @@ class Status
     attr_reader :termsig
     attr_reader :stopsig
 
-    def initialize(pid=nil, status=nil, termsig=nil, stopsig=nil)
+    def initialize(pid=nil, exitstatus=nil, termsig=nil, stopsig=nil, raw_status=nil)
       @pid = pid
-      @status = status
+      @exitstatus = exitstatus
       @termsig = termsig
       @stopsig = stopsig
+      @raw_status = raw_status
     end
 
     def exitstatus
-      @status
+      @exitstatus
     end
 
     def to_i
-      @status
+      @raw_status
     end
 
     def &(num)
-      @status & num
+      @raw_status & num
     end
 
     def ==(other)
       other = other.to_i if other.kind_of? Process::Status
-      @status == other
+      @raw_status == other
     end
 
     def >>(num)
-      @status >> num
+      @raw_status >> num
     end
 
     def coredump?
       false
     end
 
     def exited?
-      @status != nil
+      @raw_status != nil
```

Comment:
This one should be `@exitstatus`

Reply:
I'll fix those two :)

Reply:
Thanks. We will need test coverage on those as well.

Reply:
Good point, actually there are already specs but one of the `exited?` specs was tagged. That one fails with `@raw_status`.
`@raw_status` seems to work in `success?` but I think it's just luck of the native encoding for exited.

Reply:
Together with #1790 we pass all Process::Status specs.